src/HasDataCapableTrait.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Dhii\Data\Object;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+
+/**
+ * Functionality for data checking.
+ *
+ * @since [*next-version*]
+ */
+trait HasDataCapableTrait
+{
+    /**
+     * Check data by key.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $key The key, for which to check the data.
+     *
+     * @return bool True if data for the specified key exists; false otherwise.
+     */
+    protected function _hasData($key)
+    {
+        if (!is_null($key) && !is_string($key) && !($key instanceof Stringable)) {
+            throw $this->_createInvalidArgumentException($this->__('Data key must be stringable'), null, null, $key);
+        }
+
+        $store = $this->_getDataStore();
+        $key   = (string) $key;
+
+        return property_exists($store, $key);
+    }
+
+    /**
+     * Retrieves a pointer to the data store.
+     *
+     * @since [*next-version*]
+     *
+     * @return mixed The data store.
```

Comment:
Perhaps this should type-hint a return type of `object` since `_hasData()` expect an object.